### PR TITLE
[TensorRT] adapt for TRT lib name change after TRT 10 GA

### PR DESCRIPTION
**Pull request by chilo-ms**
For TensorRT 10 GA onwards, the TensorRT libraries will have major version appended to the end on Windows, for example, nvinfer_10.dll, nvinfer_plugin_10.dll, nvonnxparser_10.dll ...

Change cmake file accordingly.

### CODE REVIEW

File: cmake/onnxruntime_providers_tensorrt.cmake [hunk 1]
```diff
@@ -40,15 +40,74 @@
...
+      MESSAGE(STATUS "Can't find ${NVINFER_PLUGIN_LIB}")
+    endif()
+
+    find_library(TENSORRT_LIBRARY_NVONNXPARSER ${PARSER_LIB}
       HINTS  ${TENSORRT_ROOT}
       PATH_SUFFIXES lib lib64 lib/x64)
+
+    if (NOT TENSORRT_LIBRARY_NVONNXPARSER)
+      MESSAGE(STATUS "Can't find ${PARSER_LIB}")
+    endif()
```
Comment (kevinch-nv) [inline cmake/onnxruntime_providers_tensorrt.cmake:109]: Need to update the parser library names on lines 135-139 as well

Comment (chilo-ms) [reply cmake/onnxruntime_providers_tensorrt.cmake:109]: thanks, updated

**Review by jywu-msft — APPROVED** [review]

File: cmake/onnxruntime_providers_tensorrt.cmake [hunk 2]
```diff
@@ -73,11 +132,11 @@
...
       unset(OLD_CMAKE_CUDA_FLAGS)
-      set_target_properties(nvonnxparser PROPERTIES LINK_FLAGS "/ignore:4199")
-      target_compile_options(nvonnxparser_static PRIVATE /FIio.h /wd4100)
-      target_compile_options(nvonnxparser PRIVATE /FIio.h /wd4100)
+      set_target_properties(${PARSER_LIB} PROPERTIES LINK_FLAGS "/ignore:4199")
+      target_compile_options(${PARSER_LIB}_static PRIVATE /FIio.h /wd4100)
+      target_compile_options(${PARSER_LIB} PRIVATE /FIio.h /wd4100)
     endif()
-    set(onnxparser_link_libs nvonnxparser_static)
+    set(onnxparser_link_libs ${PARSER_LIB}_static)
```
Comment (jywu-msft) [inline cmake/onnxruntime_providers_tensorrt.cmake:139]: @kevinch-nv would the static lib built using oss parser have the version in it as well? i.e. nvonnxparser_10_static.lib ? that feels a bit unconventional.

Comment (kevinch-nv) [reply cmake/onnxruntime_providers_tensorrt.cmake:139]: Good catch. Static libraries are just `nvonnxparser_static` on all platforms.

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**